tests/csapi/upload_keys_test.go

```diff
@@ -174,6 +175,45 @@ func TestUploadKey(t *testing.T) {
 	})
 }
 
+// Per MSC4225, keys must be issued in the same order they are uploaded
+func TestKeyClaimOrdering(t *testing.T) {
+	deployment := complement.Deploy(t, 1)
+	defer deployment.Destroy(t)
+	alice := deployment.Register(t, "hs1", helpers.RegistrationOpts{})
+	_, oneTimeKeys := alice.MustGenerateOneTimeKeys(t, 2)
+
+	// first upload key 1, sleep a bit, then upload key 0.
+	otk1 := map[string]interface{}{"signed_curve25519:1": oneTimeKeys["signed_curve25519:1"]}
+	alice.MustDo(t, "POST", []string{"_matrix", "client", "v3", "keys", "upload"},
+		client.WithJSONBody(t, map[string]interface{}{"one_time_keys": otk1}))
+
+	time.Sleep(1 * time.Second)
```

Comment:
Why do we need to sleep a bit? We process OTKs asyncly in synapse?

Reply:
oh, this was to make sure there was a difference in timestamp between the two upload requests. I should probably comment to that effect.


```suggestion
    // Ensure that there is a difference in timestamp between the two upload requests.
	time.Sleep(1 * time.Second)
```